### ISSUE TITLE
Define schema for bakeries.yaml

### ISSUE THREAD

**Issue by rabernat** [pangeo-forge/bakery-database#5]
The yaml file in this repo is not fully aligned with https://github.com/pangeo-forge/roadmap/blob/master/doc/adr/0004-use-yaml-file-for-bakery-database.md. That's okay. But we need to get them consistent.

Ideally we would have some sort of formal or informal schema that enumerates all the possible options in this file.

If we had a formal scheme, we could also do schma validation via CI. At the least we could be linting the yaml.

**Comment by cisaacstern** [comment]
I agree we need this: input validation for https://github.com/pangeo-forge/pangeo-forge-orchestrator/pull/1 is one pressing use case.

Looks like @sharkinsspatial's work on [pangeo_forge_prefect/meta_types/bakery.py](https://github.com/pangeo-forge/pangeo-forge-prefect/blob/master/pangeo_forge_prefect/meta_types/bakery.py) provides a strong basis for it.

I'm going to merge that module into the CLI PR and see about refactoring it with Pydantic for validation. From there, we can [generate a JSON schema](https://pydantic-docs.helpmanual.io/usage/schema/) if desired.

**Comment by cisaacstern** [comment]
Wanted to note that https://github.com/pangeo-forge/pangeo-forge-orchestrator/pull/1 now defines a [`BakeryDatabase` pydantic model](https://github.com/pangeo-forge/pangeo-forge-orchestrator/blob/8eceefec78e0dc1932399deccf72d71b9682c540/pangeo_forge_orchestrator/meta_types/bakery.py#L201-L223) and associated [validation function](https://github.com/pangeo-forge/pangeo-forge-orchestrator/blob/minimal-cli/pangeo_forge_orchestrator/validation/validate_bakery_database.py).

> The yaml file in this repo is not fully aligned with https://github.com/pangeo-forge/roadmap/blob/master/doc/adr/0004-use-yaml-file-for-bakery-database.md.

This is borne out [in the tests here](https://github.com/pangeo-forge/pangeo-forge-orchestrator/blob/8eceefec78e0dc1932399deccf72d71b9682c540/tests/test_validation.py#L18-L20), which show that the call to `validate_bakery_database()` without kwargs (which implicitly validates the existing https://github.com/pangeo-forge/bakery-database/blob/main/bakeries.yaml) raises a validation error.

As such, we actually _can't_ use the proposed CLI to introspect our actual database just yet, because the CLI also requires inputs to be parsable into the `BakeryDatabase` model. I'm aiming to have a review-ready draft of the CLI PR (which is really a data model PR...) shortly. Once it's reviewed & merged, we can circle back to this issue and bring our existing bakery database within the spec.
